### ISSUE TITLE
A GUI with seeking bar?

### ISSUE THREAD

**Issue by rayman3003** [libsidplayfp/sidplayfp#6]
Is it possible to add a simple GUI for sidplayfp?

It is more user-friendly for normal people & also having a "seeking bar" for that GUI, makes it even better.

**Comment by drfiemost** [comment]
There are already GUI applications with libsidplayfp support available such as [audacious](https://audacious-media-player.org/), [SIDex](http://support.xmplay.com/files_view.php?file_id=720) for XMPlay, [qmmp](http://qmmp.ylsoftware.com/), [foobar](https://www.foobar2000.org), [xsidplay2](https://sourceforge.net/projects/xsidplay2/), ...
This is only a simple console showcase app and always will be.

**Comment by rayman3003** [comment]
> There are already GUI applications with libsidplayfp support available such as [audacious](https://audacious-media-player.org/), [SIDex](http://support.xmplay.com/files_view.php?file_id=720) for XMPlay, [qmmp](http://qmmp.ylsoftware.com/), [foobar](https://www.foobar2000.org), [xsidplay2](https://sourceforge.net/projects/xsidplay2/), ...

I tried all of these players (1by1) & beside foobar, none of them support "seeking bar" for sid files.

And the sidplay "seeking bar" of foobar has a long 4-5 seconds **delay**, that makes it very unpleasant.

Do you know any player that uses libsidplayfp & support **instant** "seeking bar" for sid files?

**Comment by drfiemost** [comment]
Well, sid file are not streams but programs that run in an emulated environment, "seeking" is achieved by speeding up the emulation to move quickly forward in time, but it's limited by computing power and it cannot be instantaneous.
The only workaround is to convert the file into a wav or other seekable format.

**Comment by rayman3003** [comment]
> 
> 
> Well, sid file are not streams but programs that run in an emulated environment, "seeking" is achieved by speeding up the emulation to move quickly forward in time, but it's limited by computing power and it cannot be instantaneous.
> The only workaround is to convert the file into a wav or other seekable format.

That's not true. I already found 2 solutions to play sid files **with** instant seeking.

One of them was an old Winamp plugin (I think it's name was **sidamp**). But bcuz it was old & had some serious bugs, I ditched it.

Another one that I recently found is [**ACID 64 Player**](https://www.acid64.com/). It's good, but it lacks some features.

So I tried to search & found your solution "sidplayfp". I tried to play sid files with sidplayfp & it sounds good; But it doesn't support "seeking".

Those two apps shows that it's possible to add "seeking" for playing-back "sid" files with normal computers. (my computer is an old pc!)

So if you just add "seeking" to sidplayfp, we could have a pleasant experience with sid files on our own computers. sid tunes are great to listen. :)
